### PR TITLE
[libc] newheadergen: added assert.yaml

### PR DESCRIPTION
**Pull request by aaryanshukla**
- removed assert macro definitions in api.td
- included macro definitions in assert.h.def
- added assert.yaml

### CODE REVIEW

**Review by michaelrj-google — APPROVED** [review]
LGTM, though if it breaks the GPU build we may have to revert for @jhuber6

**Comment by petrhosek** [comment]
This broke all our embedded projects that use LLVM libc because `static_assert` is now only available when `NDEBUG` isn't set. I have uploaded #99742 as fix forward but if we cannot merge it quickly I'll need to revert this change.

**Pull request merged**
